src/Orleans/Core/GrainIdentity.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using Orleans.Runtime;
+
+namespace Orleans.Core
+{
+    internal class GrainIdentity : IGrainIdentity
```

Comment:
can we get rid of this class and make GrainId implement IGrainIdentity directly? 
That way we can also save the extra code in GrainExtensions where we now have 2 different ways get identity: via IGrainIdentity (for IGrain) and via GrainId (for IAddressable).